generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs

```diff
@@ -121,66 +121,76 @@ import { mapIdList } from 'app/shared/util/entity-utils';
 
 export interface I<%= entityReactName %>UpdateProps extends StateProps, DispatchProps, RouteComponentProps<{id: string}>  {}
 
-export interface I<%= entityReactName %>UpdateState {
-  isNew: boolean;
+export const <%= entityReactName %>Update = (props: I<%= entityReactName %>UpdateProps) => {
   <%_ manyToManyOwners.forEach(val => { _%>
-  ids<%= val.relationshipName %>: any[];
+  const [ids<%= val.relationshipName %>, setIds<%= val.relationshipName %>] = useState([]);
   <%_ }) _%>
   <%_ relFieldNames.forEach(val => { _%>
-  <%= val.relationshipFieldName %>Id: string;
+  const [<%= val.relationshipFieldName %>Id, set<%= upperFirstCamelCase(val.relationshipFieldName) %>Id] = useState('0');
   <%_ }) _%>
-}
-
-export class <%= entityReactName %>Update extends React.Component<I<%= entityReactName %>UpdateProps, I<%= entityReactName %>UpdateState> {
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      <%_ manyToManyOwners.forEach(val => { _%>
-      ids<%= val.relationshipName %>: [],
-      <%_ }) _%>
-      <%_ relFieldNames.forEach(val => { _%>
-      <%= val.relationshipFieldName %>Id: '0',
-      <%_ }) _%>
-      isNew: !this.props.match.params || !this.props.match.params.id,
-    };
-  }
+  const [isNew, setIsNew] = useState(!props.match.params || !props.match.params.id);
```

Comment:
I'd have named these this way `const [newEntity, setIsNewEntity]`, because `isNew` looks like a function's name and not a variable's name.